core/domain/draft_upgrade_services.py

```diff
@@ -88,6 +88,70 @@ def try_upgrading_draft_to_exp_version(
 class DraftUpgradeUtil(python_utils.OBJECT):
     """Wrapper class that contains util functions to upgrade drafts."""
 
+    @classmethod
+    def _convert_states_v30_dict_to_v31_dict(cls, draft_change_list):
+        """Converts draft change list from state version 30 to 31. State
+        Version 31 adds the duration float for the Voiceover section of state.
+
+        Args:
+            draft_change_list: list(ExplorationChange). The list of
+                ExplorationChange domain objects to upgrade.
+
+        Returns:
+            list(ExplorationChange). The converted draft_change_list.
+        """
+        for i, change in enumerate(draft_change_list):
+            if (change.cmd == exp_domain.CMD_EDIT_STATE_PROPERTY and
+                    change.property_name ==
+                    exp_domain.STATE_PROPERTY_RECORDED_VOICEOVERS):
+                # Get the language code to access the language code correctly.
+                new_change = change.new_value['voiceovers_mapping']
+                language_code = new_change['content'].keys()
+                # Initialize the value to migrate draft state to v31.
+                new_change['content'][language_code[0]]['duration'] = 0.0
+                draft_change_list[i] = exp_domain.ExplorationChange({
+                    'cmd': exp_domain.CMD_EDIT_STATE_PROPERTY,
+                    'property_name': (
+                        exp_domain.STATE_PROPERTY_RECORDED_VOICEOVERS),
+                    'state_name': change.state_name,
+                    'new_value': {
+                        'voiceovers_mapping': new_change
+                    }
+                })
+        return draft_change_list
+
+
+    @classmethod
+    def _convert_states_v29_dict_to_v30_dict(cls, draft_change_list):
```

Comment:
There's already a _convert_states_v29_dict_to_v30_dict method below, why are we overwriting it here?

Reply:
Whoops I'll get rid of it. My bad.

Reply:
Done.